packages/backend/src/accounting/tigerbeetle/accounts.ts

```diff
@@ -50,12 +54,24 @@ export async function createAccounts(
   }
 }
 
+export function flagsBasedOnAccountOptions(
+  options: CreateAccountOptions
+): AccountFlags {
+  let returnVal =
+    options.code === TigerBeetleAccountCode.SETTLEMENT
+      ? AccountFlags.credits_must_not_exceed_debits
+      : AccountFlags.debits_must_not_exceed_credits
+  if (options.linked) returnVal |= AccountFlags.linked
+  if (options.history) returnVal |= AccountFlags.history
+  return returnVal
+}
+
```

Comment:
Just reading this and finding it a bit confusing. Not saying there is anything wrong with it necessarily, and I realize this probably traces back to simply how the tb node client works, but I'm just not sure how this is supposed to work. Can you add a unit test for this - I think that would go a long way in clarifying how this is expected to work.

As I understand it, if we call `flagsBasedOnAccountOptions` with options:

```ts
{
  code: TigerBeetleAccountCode.SETTLEMENT, 
  linked: true, 
  history: true
}
````

Then the `returnValue` is  `13`, but this doesnt make sense to me because the return type is `AccountFlags` and `13` is not one of the values for this `AccountFlags` enum.

Let me show my math.

| Variable            | Value                            |
|---------------------|----------------------------------|
| `AccountFlags.credits_must_not_exceed_debits`      | `4` |
| `AccountFlags.linked`    | `1`                           |
| `AccountFlags.history`   | `8`                          |

So it the function in this case basically resolves to this (you can copy/paste this into a js/node interpreter to verify):

```js
returnVal = 4 //  let returnVal = options.code === TigerBeetleAccountCode.SETTLEMENT ? AccountFlags.credits_must_not_exceed_debits : AccountFlags.debits_must_not_exceed_credits
returnVal |= 1 // if (options.linked) returnVal |= AccountFlags.linked
returnVal |= 8 // if (options.history) returnVal |= AccountFlags.history

returnVal // resolves to 13
```

This just doesnt seem to add up because the return value is supposed to be of type `AccountFlags`

Reply:
@BlairCurrey that is correct. ✅ I have added test cases with explanations to clarify. The TigerBeetle `flags` field is a unsigned 16bit field. The bitfield is used to flag none, one or many different "feature" flags. The resulting `flags` value will depend on what combination of flags need to be active.
The "opposite" is true (using the `&` operator) for when you want to verify if a certain flag/feature is enabled. Example:
```
const credExcDebitAndHistory = 12
    expect(
      flagsBasedOnAccountOptions({
        id: uuid(),
        code: TigerBeetleAccountCode.SETTLEMENT,
        ledger: 0,
        userData128: 0n,
        history: true
      })
    ).toEqual(credExcDebitAndHistory)

    // Tests to see if correct flags are set.
    expect(
      AccountFlags.credits_must_not_exceed_debits & credExcDebitAndHistory
    ).toBeTruthy()
expect(AccountFlags.linked & credExcDebitAndHistory).toBeFalsy()
```

Please let me know if it all makes sense? 


Basically the Linux file permissions work similar:
<img width="767" alt="image" src="https://github.com/user-attachments/assets/3fe43838-95fd-4785-947a-4fbe1f392ab3">



https://github.com/interledger/rafiki/pull/2776/files#r1686518809